nodebuilder/header/config.go

```diff
@@ -82,34 +82,44 @@ func (cfg *Config) trustedPeers(bpeers p2p.Bootstrappers) (infos []peer.AddrInfo
 	return
 }
 
-func (cfg *Config) trustedHash(net p2p.Network) (libhead.Hash, error) {
-	if cfg.TrustedHash == "" {
-		gen, err := p2p.GenesisFor(net)
-		if err != nil {
-			return nil, err
-		}
-		return hex.DecodeString(gen)
-	}
-	return hex.DecodeString(cfg.TrustedHash)
-}
-
 // Validate performs basic validation of the config.
 func (cfg *Config) Validate(tp node.Type) error {
 	err := cfg.Store.Validate()
 	if err != nil {
 		return fmt.Errorf("module/header: misconfiguration of store: %w", err)
 	}
 
-	err = cfg.Syncer.Validate()
-	if err != nil {
-		return fmt.Errorf("module/header: misconfiguration of syncer: %w", err)
-	}
-
 	err = cfg.Server.Validate()
 	if err != nil {
 		return fmt.Errorf("module/header: misconfiguration of p2p exchange server: %w", err)
 	}
 
+	if cfg.Syncer.TrustingPeriod != trustingPeriod {
```

Comment:
I think we shouldn't expose this parameter, the same way we don't expose the sampling window